### DISCUSSION TITLE
Can I use Kymatio wavelet scattering as a generative differentiable model

### DISCUSSION THREAD

**Discussion by jakubMitura14** [kymatio/kymatio#1000] [Q&A] [answered]
Hello I am working on a model where one part is responsible for texture generation.
My idea is that I have set of learnable parameters (scattering coefficients) that will be used to generate texture than the output texture will have some loss function. Task is to create a texture with minimal loss function using gradient descent based  optimazation on scattering parameters.
Work is on 3d texture in medical imaging
Is it achievable?

**Comment by lostanlen** [comment, marked as answer]
Hello @jakubMitura14, yes this has been done in 1-D and 2-D in the past so presumably it is also achievable in 3D. Let us know how you make progress with it

**Reply by jakubMitura14** [reply]
Thanks!! Ok so I am starting working on it, in case of big problems will get back here

**Reply by jakubMitura14** [reply]
Hello @lostanlen Hovewer I would like to be sure is it possible to perform a deterministic inverse transform - so given scattering coefficients recreate an image, sth like inverse Fourier transform? Because as far as I am analyzing the examples given all inverses are learned via optimization.

**Reply by lostanlen** [reply]
@jakubMitura14 there is no known closed-form pseudo-inverse for the scattering transform. The simplest solution (as you've seen in the tutorial) is to solve the inverse problem by iterative gradient-based optimization. A more advanced solution is to learn the pseudo-inverse operator as a generative model. In that case, resynthesis can happen in a single step, without need to run an inverse problem solver.

**Reply by jakubMitura14** [reply]
Thank you for your time and for sharing your expertise!
